From 5414a1138b1d3fb5dc20f1c832b0eb0e4270c88e Mon Sep 17 00:00:00 2001
From: Weifeng Voon <weifeng.voon@intel.com>
Date: Fri, 17 Jun 2016 16:39:18 +0800
Subject: [PATCH 2245/2508] i2c: designware: Fallback to fast speed if high
 speed isn't supported

commit 50f06a9e0b6ece8c9ad037e8f4df1645b247590f from
https://github.com/01org/linux-apollolake-i

When high speed is not supported, the sda-hold-time will fall back to use
FMCN parameters if it is available.

Signed-off-by: Weifeng Voon <weifeng.voon@intel.com>
---
 drivers/i2c/busses/i2c-designware-core.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/i2c/busses/i2c-designware-core.c b/drivers/i2c/busses/i2c-designware-core.c
index d889604..f421801 100644
--- a/drivers/i2c/busses/i2c-designware-core.c
+++ b/drivers/i2c/busses/i2c-designware-core.c
@@ -383,6 +383,8 @@ int i2c_dw_init(struct dw_i2c_dev *dev)
 			dev_err(dev->dev, "High Speed not supported!\n");
 			dev->master_cfg &= ~DW_IC_CON_SPEED_MASK;
 			dev->master_cfg |= DW_IC_CON_SPEED_FAST;
+			if (dev->fs_sda_ht)
+				dev->sda_hold_time = dev->fs_sda_ht;
 		} else if (dev->hs_hcnt && dev->hs_lcnt) {
 			hcnt = dev->hs_hcnt;
 			lcnt = dev->hs_lcnt;
-- 
1.7.5.4

